File: back-end/MS-Auth/auth-service/src/main/java/com/bantads/auth_service/DTOs/GerenteEvent.java
package com.bantads.auth_service.DTOs;

public record GerenteEvent(
    String tipo,
    String cpfGerente,
    String nome,
    String email,
    String senha,
    String cpfNovoGerente,
    String cpfGerenteAnterior,
    String numeroConta
) {}

File: back-end/MS-Gerente/gerente-service/src/main/kotlin/com/bantads/gerente_service/dto/GerenteEvent.kt
package com.bantads.gerente_service.dto

/*Evento publicado pelo MS-Gerente para integracao via RabbitMQ*/
data class GerenteEvent(
    /*insercao, alteracao ou remocao*/
    val tipo: String,

    /*CPF do gerente principal do evento*/
    val cpfGerente: String,

    /*Dados usados principalmente em insercao/alteracao*/
    val nome: String? = null,
    val email: String? = null,
    val senha: String? = null,

    /*Gerente que deve assumir contas em uma remocao*/
    val cpfNovoGerente: String? = null,

    /*Campos para eventos de transferencia de conta*/
    val cpfGerenteAnterior: String? = null,
    val numeroConta: String? = null
)
